gpii/node_modules/transformer/src/js/Transformer.js

```diff
@@ -79,6 +79,10 @@ var fluid = fluid || require("infusion"),
         var togo = {};
         fluid.each(configuration.applications, function (preferences, solutionId) {
             togo[solutionId] = fluid.copy(solutionsRegistry[solutionId]);
+            if (togo[solutionId] === undefined) {
+                fluid.fail("The configuration to be applied contains an entry for " + solutionId +
```

Comment:
This uses the wrong error signalling scheme - fluid.fail will in general issue an uncaught exception for which the behaviour of the system following will be undefined. This should be reserved for coding errors for which one, for example, in C/C++ would have issued an "assert". An error which merely represents an error in a payload or persistent data in the system, which is expected to be recoverable if the system receives a different request should be signalled by returning a rejected promise.

Note that this pull will conflict with the GPII-17 branch which renames and reorganises many of these functions